crates/biome_fs/src/path.rs

```diff
@@ -3,34 +3,106 @@
 //! give additional information around the file that holds:
 //! - the [FileHandlers] for the specific file
 //! - shortcuts to open/write to the file
+use crate::ConfigName;
+use enumflags2::{bitflags, BitFlags};
+use std::cmp::Ordering;
 use std::ffi::OsStr;
-use std::fs::{self, read_to_string};
+use std::fs::read_to_string;
+use std::hash::Hash;
+use std::path::Path;
 use std::{fs::File, io, io::Write, ops::Deref, path::PathBuf};
 
-#[derive(Debug, Clone, Eq, Hash, PartialEq, Default)]
+/// The priority of the file
+#[derive(Debug, Clone, Copy, Default, PartialEq, Eq, Ord, PartialOrd, Hash)]
+#[repr(u8)]
+#[bitflags]
 #[cfg_attr(
     feature = "serde",
     derive(serde::Serialize, serde::Deserialize, schemars::JsonSchema)
 )]
-pub struct BiomePath {
-    path: PathBuf,
+// NOTE: The order of the variants is important, the one on the top has the highest priority
+pub enum FileKind {
+    /// A configuration file has the highest priority. It's usually `biome.json` and `biome.jsonc`
+    ///
+    /// Other third-party configuration files might be added in the future
+    Config,
+    /// It's usually `package.json` and `tsconfig.json`
+    Manifest,
+    /// An ignore file, like `.gitignore`
+    Ignore,
+    /// Files that are required to be inspected before handling other files.
+    ///
+    /// An example is the GraphQL schema
+    ToInspect,
```

Comment:
It feels a bit off to use a verb with `To` prefix for these cases. How about `Inspectable` and `Handleable`?